### PR TITLE
MOB-629

### PR DESCRIPTION
**Pull request by shekhardahore**
Updated liRepliesClient to handle offset for messages.

### CODE REVIEW

File: Sources/LiCore/QueryUtil/LiQueryBuilder.swift [hunk 1]
```diff
@@ -105,6 +106,12 @@ public class LiQueryBuilder {
             query.append(space)
             query.append(limitSetting)
         }
+        if let offsetSetting = liQuerySetting.offset {
+            query.append(space)
+            query.append(offset)
+            query.append(space)
+            query.append(offsetSetting)
```
Comment (AatishMolasi) [inline Sources/LiCore/QueryUtil/LiQueryBuilder.swift]: Isn't is possible to just format it using " \(offset) \(offsetSetting)" ?

File: Sources/LiCore/QueryUtil/LiQuerySetting.swift [hunk 2]
```diff
@@ -96,6 +98,11 @@ public class LiQuerySetting {
         } else {
             self.limit = nil
         }
+        if let offset = data["offset"] as? String {
+            self.offset = offset
+        } else {
+            self.offset = nil
+        }
```
Comment (AatishMolasi) [inline Sources/LiCore/QueryUtil/LiQuerySetting.swift]: Couldn't you write this as ..
`self.offset = data["offset"] as? String`

File: Sources/LiCore/Networking/LiClient.swift [hunk 3]
```diff
@@ -288,7 +288,7 @@ extension LiClient {
         case .liUploadImageClient:
             return [:]
         case .liRepliesClient(let requestParams):
-            return [LiQueryConstant.liMarkAsRead: "true", "q": LiUrlConstructor.getLiqlQuery(client: self).replacingOccurrences(of: "##", with: requestParams.parentId)]
+            return [LiQueryConstant.liMarkAsRead: "true", "q": LiUrlConstructor.getLiqlQuery(client: self).replacingOccurrences(of: "##", with: requestParams.parentId).replacingOccurrences(of: "**", with: "\(requestParams.limit)").replacingOccurrences(of: "&&", with: "\(requestParams.offset)")]
```
Comment (AatishMolasi) [inline Sources/LiCore/Networking/LiClient.swift]: are you sure the other params will not contain these chars (** ..etc)?

File: Sources/LiCore/QueryUtil/LiQueryBuilder.swift (hunk 1 above)
Comment (shekhardahore) [reply Sources/LiCore/QueryUtil/LiQueryBuilder.swift]: Yes, this is to keep it a bit clearer.

File: Sources/LiCore/QueryUtil/LiQuerySetting.swift (hunk 2 above)
Comment (shekhardahore) [reply Sources/LiCore/QueryUtil/LiQuerySetting.swift]: Will update this

File: Sources/LiCore/Networking/LiClient.swift (hunk 3 above)
Comment (shekhardahore) [reply Sources/LiCore/Networking/LiClient.swift]: Yes

File: Sources/LiCore/QueryUtil/LiQueryBuilder.swift (hunk 1 above)
Comment (AatishMolasi) [reply Sources/LiCore/QueryUtil/LiQueryBuilder.swift]: I personally think appending it like this is a little more difficult to read and also more prone to crashes but I'll leave it upto you.

File: Sources/LiCore/Networking/LiClient.swift (hunk 3 above)
Comment (AatishMolasi) [reply Sources/LiCore/Networking/LiClient.swift]: Also, in the future instead of using stringByReplacingString check out string formatter.

Comment (shekhardahore) [reply Sources/LiCore/Networking/LiClient.swift]: aye aye

**Pull request merged**